src/ScoreTrack.py

```diff
@@ -287,84 +375,130 @@ def initialize_leaderboard():
+                new_score = {"initials": initials, "full_name": "    ", "score": score, "date": f"{month:02d}/{day:02d}/{year}", "game_count": SharedState.gameCounter}
+                update_leaderboard(new_score)
+
+
+def update_tournamentboard(new_entry):
+    """
+    place a new score in the tournament board fram
+
+    """
+    count = DataStore.memory_map["tournament"]["count"]
+    rec = DataStore.read_record("tournament", 0)
+    nextIndex = rec["index"]
+
+    new_entry["game"] = SharedState.gameCounter
+    new_entry["index"] = nextIndex
+    DataStore.write_record("tournament", new_entry, nextIndex)
+    print("SCORE: tournament new score ", new_entry)
+
+    nextIndex += 1
+    if nextIndex >= count:
+        nextIndex = 0
+    rec = DataStore.read_record("tournament", 0)
+    rec["index"] = nextIndex
+    DataStore.write_record("tournament", rec, 0)
+    return
+
+
+def place_game_in_tournament(game):
+    """
+    place game (all four scores) in tournament board
+        game: gamecounter, score 0, 1, 2, 3
+    """
+    year, month, day, _, _, _, _, _ = rtc.datetime()
+    new_score = {"initials": " ", "full_name": " ", "score": 0, "date": f"{month:02d}/{day:02d}/{year}", "game": game[0]}
+
     for i in range(4):
-        initials, score = readMachineScore(i)
-        if any(0x40 < ord(initial) <= 0x5A for initial in initials[:3]):
-            year, month, day, _, _, _, _, _ = rtc.datetime()
-            new_score = {
-                "initials": initials,
-                "full_name": "    ",
-                "score": score,
-                "date": f"{month:02d}/{day:02d}/{year}",
-                "game_count": SharedState.gameCounter,
-            }
-            update_leaderboard(new_score)
+        if int(game[i + 1][1]) > 0:
+            new_score["score"] = int(game[i + 1][1])
+            update_tournamentboard(new_score)
+
+
+def place_game_in_claim_list(game):
+    recent_scores[5] = recent_scores[4]
+    recent_scores[4] = recent_scores[3]
+    recent_scores[3] = recent_scores[2]
+    recent_scores[2] = recent_scores[1]
+    recent_scores[1] = recent_scores[0]
+    recent_scores[0] = game
+    print("SCORE: add to claims list: ", recent_scores)
+
+
+nGameIdleCounter = 0
 
 
 def CheckForNewScores(nState=[0]):
-    enscorecap = DataStore.read_record("extras", 0)["other"]
-    if not bool(enscorecap) or S.gdata["HighScores"]["Type"] == 0:
-        return
+    """
+    called by server
+        call rate is every ~5 seconds
+
+    system 11:
+        if enter_initials_on_game is on, remove high scores during game to force all player to enter intiials
+
+    system 9:
+        if display ip address is on, do not place high scores in machine memeory (same display)
+        enter_initials_on_game is not supported for system 9
+
+    """
+    global nGameIdleCounter
+
+    # TODO unused variable that maybe should be used somewhere?
+    # enscorecap = DataStore.read_record("extras", 0)["enter_initials_on_game"]
```

Comment:
This variable wasn't being used. But it seems like maybe it should be? I commented it out for now.